sntpc/Cargo.toml

```diff
@@ -28,13 +28,15 @@ log = ["dep:log"]
 std-socket = []
 embassy-socket = ["dep:embassy-net"]
 tokio-socket = ["dep:tokio"]
+defmt = ["dep:defmt", "embassy-net/defmt"]
 
 [dependencies]
 log = { version = "~0.4", optional = true }
 chrono = { version = "~0.4", default-features = false, optional = true }
 miniloop = { version = "~0.3", optional = true }
 embassy-net = { version = "0.5.0", features = ["udp", "proto-ipv4", "proto-ipv6", "medium-ip"], optional = true }
 tokio = { version = "1", features = ["net"], optional = true }
+defmt = { version = "0.3", optional = true }
```

Comment:
I assume that after introducing that feature we have two mutually exclusive features: `defmt` and `log`.
It is worth adding the following to prevent user by enabling both:
```rust
#[cfg(all(feature = "log", feature = "defmt"))]
compile_error!("Having both `defmt` and `log` features enabled is not supported");
```

Reply:
Quoting the Rust book

```
There are rare cases where features may be mutually incompatible with one another. This should be avoided if at all possible, because it requires coordinating all uses of the package in the dependency graph to cooperate to avoid enabling them together.
```

Basically this doesn't play well at all with `--all-features`